### ISSUE TITLE
Not possible to use server on HTTPS with a valid public certificate

### ISSUE THREAD

**Issue by gauthierm** [pksprojects/ElasticSwift#95]
**Describe the bug**
On iOS (and possibly macOS) The custom self-signed TLS certificate support doesn't let you use a valid public certificate from—for example—Let's Encrypt.

**To Reproduce**
Trying to use the example from the documentation to make a basic query on a server that is using a valid public TLS certificate on iOS:

```swift
import ElasticSwift
import ElasticSwiftNetworking

var settings = Settings(forHost: "https://www.example.com", adaptorConfig: URLSessionAdaptorConfiguration.default) // Creates default settings for client
var client = ElasticClient(settings: settings) // Creates client with specified settings
```

This causes a relatively opaque error message:

```
Error Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLStringKey=https://www.example.com/my-index/_search?, NSErrorFailingURLKey=https://www.example.com/my-index/_search?, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <CD9D4F4D-A6F9-4BA9-8B29-E75A486DFA3E>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <CD9D4F4D-A6F9-4BA9-8B29-E75A486DFA3E>.<1>, NSLocalizedDescription=cancelled}
```

**Expected behavior**
I expect that supporting a self-signed certificate requires configuration but supporting a public trusted certificate works by default. The NIO adapter seems to work by default.

**Environment Details (please complete the following information):**
 - OS: iOS
 - Version iOS 14, Xcode 12

**ElasticSwift (please complete the following information):**
 - Version 1.0.0-beta.1

**Additional context**
It looks to me like the code in https://github.com/pksprojects/ElasticSwift/blob/master/Sources/ElasticSwiftNetworking/SessionManager.swift#L94-L96 is what causes the -999 error to occur. I'm not familiar enough with custom TLS handling in URLSession to be able to say for sure, but should this call the completion handler with `.performDefaultHandling` instead of `.cancelAuthenticationChallenge`?

**Comment by prafsoni** [comment]
PR for fix #97

**Comment by prafsoni** [comment]
#97 was merged
